pkg/storage/stores/shipper/bloomshipper/store.go

```diff
@@ -0,0 +1,356 @@
+package bloomshipper
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"path/filepath"
+	"sort"
+
+	"github.com/go-kit/log"
+	"github.com/pkg/errors"
+	"github.com/prometheus/common/model"
+
+	"github.com/grafana/loki/pkg/storage"
+	"github.com/grafana/loki/pkg/storage/chunk/cache"
+	"github.com/grafana/loki/pkg/storage/chunk/client"
+	"github.com/grafana/loki/pkg/storage/config"
+)
+
+type Store interface {
+	SearchMetas(ctx context.Context, params MetaSearchParams) ([]Meta, error)
```

Comment:
Eventually, this should become:

```suggestion
	ResolveMetas(ctx context.Context, params MetaSearchParams) ([]MetaRef, error)
```

To match with

https://github.com/grafana/loki/blob/main/pkg/bloomcompactor/v2_meta.go#L125-L129